tcping.go

```diff
@@ -388,7 +392,7 @@ func getSystemTime() time.Time {
 /* Ping host, TCP style */
 func tcping(tcpStats *stats) {
 
-	IPAndPort := net.JoinHostPort(tcpStats.ip, tcpStats.port)
+	IPAndPort := net.JoinHostPort(tcpStats.ip.String(), tcpStats.port)
```

Comment:
I suggest doing the type cast/conversion inside the `resolveHostname` once. This way, we do not have to convert the IP to string whenever we need it. Although, if you forsee a future that we require it to be of type IPAddr, I am fine with it.